applications/RANSApplication/custom_processes/rans_apply_exact_nodal_periodic_condition_process.cpp

```diff
@@ -0,0 +1,321 @@
+    }
+    mRotationAngle = mrParameters["rotation_settings"]["angle"].GetDouble();
+
+    KRATOS_ERROR_IF(mRotationAngle < eps && mTranslationMagnitude < eps)
+        << "Either translation or rotation should be defined.\n";
+    KRATOS_ERROR_IF(mRotationAngle > eps && rotation_axis_norm < eps)
+        << "Rotation angle defined, but rotation axis is not defined.\n";
+    KRATOS_ERROR_IF(mTranslationMagnitude > eps && translation_direction_norm < eps) << "Translation magnitude defined, but translation direction is not defined.\n";
+
+    const int domain_size =
+        mrModel.GetModelPart(mBaseModelPartName).GetProcessInfo()[DOMAIN_SIZE];
+    KRATOS_ERROR_IF(domain_size == 2 && (mRotationAxis[0] > eps || mRotationAxis[1] > eps))
+        << "2D rotation axis should be [0.0, 0.0, 1.0].\n";
+
+    if (mRotationAngle > eps && mTranslationMagnitude > eps) {
+        noalias(mRotationCenter) += mTranslationDirection * mTranslationMagnitude;
+    }
+
+    KRATOS_CATCH("");
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::ExecuteInitialize()
+{
+    CreatePeriodicConditions();
+    if (mReorder) {
+        auto& r_model_part = mrModel.GetModelPart(mBaseModelPartName);
+        Parameters default_params(R"({})");
+        ReorderAndOptimizeModelPartProcess reorder_process(r_model_part, default_params);
+        reorder_process.Execute();
+    }
+}
+
+std::string RansApplyExactNodalPeriodicConditionProcess::Info() const
+{
+    return std::string("RansApplyExactNodalPeriodicConditionProcess");
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << this->Info();
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::PrintData(std::ostream& rOStream) const
+{
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::CreatePeriodicConditions()
+{
+    KRATOS_TRY
+
+    auto& r_base_model_part = mrModel.GetModelPart(mBaseModelPartName);
+    int condition_id = r_base_model_part.NumberOfConditions();
+    Properties::Pointer p_properties = r_base_model_part.CreateNewProperties(
+        r_base_model_part.NumberOfProperties() + 1);
+
+    auto& r_master_model_part_nodes = mrModel.GetModelPart(mMasterModelPartName).Nodes();
+    auto& r_slave_model_part_nodes = mrModel.GetModelPart(mSlaveModelPartName).Nodes();
+
+    KRATOS_ERROR_IF(r_master_model_part_nodes.size() != r_slave_model_part_nodes.size())
+        << this->Info() << " failed. Master and slave model part nodes mismatch.\n";
+
+    const int number_of_nodes = r_master_model_part_nodes.size();
+    const double eps = std::numeric_limits<double>::epsilon();
+    const array_1d<double, 3> translation = mTranslationDirection * mTranslationMagnitude;
+
+    if (mRotationAngle > eps && mTranslationMagnitude > eps) {
+#pragma omp parallel for shared(condition_id)
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node) {
+            auto& r_master_node = *(r_master_model_part_nodes.begin() + i_node);
```

Comment:
```suggestion
            auto& r_master_node = r_master_model_part_nodes.begin() + i_node;
```
I think you don't need to dereference in here.

Reply:
moved to lambda functions :), so less code duplication.